ledger/acctupdates.go

```diff
@@ -1327,23 +1327,29 @@ func (au *accountUpdates) lookupWithoutRewards(rnd basics.Round, addr basics.Add
 			return
 		}
 
+		deltas := au.deltas[:offset]
 		rewardsVersion = au.versions[offset]
 		rewardsLevel = au.roundTotals[offset].RewardsLevel
 
 		// check if we've had this address modified in the past rounds. ( i.e. if it's in the deltas )
 		macct, indeltas := au.accounts[addr]
+		if synchronized {
+			au.accountsMu.RUnlock()
+			needUnlock = false
+		}
+
 		if indeltas {
 			// Check if this is the most recent round, in which case, we can
 			// use a cache of the most recent account state.
-			if offset == uint64(len(au.deltas)) {
+			if offset == uint64(currentDeltaLen) {
```

Comment:
Note to Self: saving this value prior to this loop is helpful because it allows us to unlock sooner. And it is safe to use because `au.deltas` does not change in size over the course of the loop.